### PR TITLE
CANVAS-63 - Display reserves content.

### PR DESCRIPTION
**Pull request by kilometerskang**

### CODE REVIEW

File: src/server/models/db.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+const { MongoClient } = require('mongodb');
```
Comment (finnzink) [inline src/server/models/db.js:1]: What is the purpose of this file? Is this using MongoDB migration?

Comment (kilometerskang) [reply src/server/models/db.js:1]: it has code for connecting to the database, so code doesn't have to be repeated in queries.js

File: src/client/components/ReserveListings/index.js [hunk 2]
```diff
@@ -1,5 +1,100 @@
...
-const ReserveListings = () => <div>hi</div>;
+const ReserveListings = () => {
+  const [entries, setEntries] = useState([]);
+  const [loading, setLoading] = useState(true);
+
+  const getReserves = () => {
+    ltikPromise.then(ltik => {
+      axios
+        .get(`/api/getreserves?ltik=${ltik}`, {
+          params: { term: '20S' },
```
Comment (rlorenzo) [inline src/client/components/ReserveListings/index.js:18]: @kilometerskang This is hardcoded to use 20S. Should show all terms or a given term. See the term selector for Bruincast content at https://github.com/ucla/lti-media-resources/blob/master/src/client/components/AdminPanel/BruincastAdminListings.js#L60

File: src/client/components/AdminPanel/index.js [hunk 3]
```diff
@@ -13,8 +13,6 @@ const AdminPanel = () => {
     <Tabs
       variant="secondary"
       onRequestTabChange={(event, { index }) => setSelectedIndex(index)}
-      minHeight="10rem"
-      maxHeight="10rem"
     >
       <Tabs.Panel
         renderTitle="Course Reserve Listings"
```
Comment (rlorenzo) [inline src/client/components/AdminPanel/index.js:18]: @kilometerskang Should be renamed to "Course resource listing" to match how the other tabs are capitalized.

File: src/server/models/queries.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+require('dotenv').config();
+const e = require('express');
+const client = require('./db');
+
+const mongourl = process.env.DB_URL;
+const dbName = process.env.DB_DATABASE;
+
+module.exports.getReservesByTerm = async (dbCollection, academicTerm) => {
```
Comment (rlorenzo) [inline src/server/models/queries.js:8]: @kilometerskang Make sure that only Administrators can call this API

Comment (kilometerskang) [reply src/server/models/queries.js:8]: How do I do that?

Comment (rlorenzo) [reply src/server/models/queries.js:8]: @kilometerskang Check the user's roles in idToken: https://cvmcosta.me/ltijs/#/provider?id=idtoken

You did that on the frontend: https://github.com/ucla/lti-library-resources/blob/master/src/client/components/App/userRoles.js#L11

Comment (kilometerskang) [reply src/server/models/queries.js:8]: The AdminPanel doesn't ever get rendered unless the user is an admin. Is that what you mean, or do you mean I should pass in the idToken to the API or something?

Comment (rlorenzo) [reply src/server/models/queries.js:8]: @kilometerskang A nefarious user just call the API endpoint directly. That's why you also need to secure the backend and not just the frontend. The LTIjs backend should already have the idToken. No need to pass it back.

**Comment by finnzink** [comment]
@kilometerskang Looks good so far. Lmk if there are any updates with the Admin API problem

**Pull request merged**